cpp/src/arrow/compute/api_scalar.h

```diff
@@ -942,6 +959,21 @@ Result<Datum> ISOYear(const Datum& values, ExecContext* ctx = NULLPTR);
 /// \note API not yet finalized
 ARROW_EXPORT Result<Datum> ISOWeek(const Datum& values, ExecContext* ctx = NULLPTR);
 
+/// \brief Week returns week of year number for each element of `values`.
+/// First ISO week has the majority (4 or more) of its days in January.
+/// Year can have 52 or 53 weeks. Week numbering can start with 0 or 1
+/// depending on DayOfWeekOptions.one_based_numbering.
```

Comment:
```suggestion
/// depending on DayOfWeekOptions.count_from_zero.
```

Reply:
BTW, I find the "Week numbering can start with 0 or 1" a bit confusing, as you don't really start to count from 0 (it's not that this represents a "full" week, and that the days at the end of the year (a potential week 53) are also coded as week 0).